### ISSUE TITLE
Any Emacs shortcut for quickly generating a partial element?

### ISSUE THREAD

**Issue by ice1000** [agda/agda#6051]
It is very nice that I can refine a goal by applying the introduction rule (like when the goal is a pi type, I can press a shortcut and Emacs give me a lambda). However, this does not work for partials. I wish there can be something like this:

+ `{ i }`, refine, `λ { (i = i0) -> ? ; (i = i1) -> ? }`
+ `{ i j }`, refine, `λ { (i = i0) (j = i0) -> ? ; (i = i1) (j = i0) -> ? ; ... }`

Maybe we need a new shortcut for this because Emacs may think `i` is a function that I want to use to refine the goal, which is the intended behavior of the refine key we already have.

**Comment by gallais** [comment]
Surely it's an extension of case-splitting (`C-c C-c`)?
E.g. currently you can do

```agda
open import Agda.Builtin.Nat

t : Nat → Nat
t = λ { n → {!!} }
```
And `C-c C-c n RET` will give you back `t = λ { zero → {!!} ; (suc n) → {!!} }`.

**Comment by JacquesCarette** [comment]
I think the OP's asking for a way to generate a pattern-matching lambda. I agree that I'd really like to be able to refine-and-split in a nice way.

**Comment by ice1000** [comment]
Does the pattern matching lambda generation work for partial elements? Hmm...

**Comment by plt-amy** [comment]
No, you can't case-split on IsOne. I investigated doing so in the past but I couldn't figure out how to smuggle an equational pattern out of the coverage checker.

**Comment by jespercockx** [comment]
See also https://github.com/agda/agda/issues/4315

**Comment by plt-amy** [comment]
And #3412
